packages/ses/src/error/note-log-args.js

```diff
@@ -0,0 +1,216 @@
+ */
+const makeSelfCell = (key, value) => {
+  /** @type {DoublyLinkedCell} */
+  const selfCell = {
+    // @ts-expect-error will be fixed before return
+    next: undefined,
+    // @ts-expect-error will be fixed before return
+    prev: undefined,
+    key,
+    value,
+  };
+  selfCell.next = selfCell;
+  selfCell.prev = selfCell;
+  // Not frozen!
+  return selfCell;
+};
+
+/**
+ * Splices a self-linked non-sigil cell into a list after `prev`.
+ * `prev` could be the head sigil, or it could be some other non-sigil
+ * cell within a list.
+ *
+ * @param {DoublyLinkedCell} prev
+ * @param {DoublyLinkedCell} selfCell
+ */
+const spliceAfter = (prev, selfCell) => {
+  if (prev === selfCell) {
+    throw TypeError('Cannot splice a cell into itself');
+  }
+  if (selfCell.next !== selfCell || selfCell.prev !== selfCell) {
+    throw TypeError('Expected self-linked cell');
+  }
+  const cell = selfCell;
+  // rename variable cause it isn't self-linked after this point.
+
+  const next = prev.next;
+  cell.prev = prev;
+  cell.next = next;
+  prev.next = cell;
+  next.prev = cell;
+  // Not frozen!
+  return cell;
+};
+
+/**
+ * @param {DoublyLinkedCell} cell
+ * Must be a non-sigil part of a list, and therefore non-self-linked
+ */
+const spliceOut = cell => {
+  const { prev, next } = cell;
+  if (prev === cell || next === cell) {
+    throw TypeError('Expected non-self-linked cell');
+  }
+  prev.next = next;
+  next.prev = prev;
+  cell.prev = cell;
+  cell.next = cell;
+};
+
+/**
+ * The LRUCacheMap is used within the implementation of `assert` and so
+ * at a layer below SES or harden. Thus, we give it a `WeakMap`-like interface
+ * rather than a `WeakMapStore`-like interface. To work before `lockdown`,
+ * the implementation must use `freeze` manually, but still exhausively.
+ *
+ * It does not hold onto anything weakly. The only sense in which it is
+ * WeakMap-like is that it does not support enumeration.
+ *
+ * TODO: Make parameterized `Key` and `Value` template types
+ *
+ * @param {number} budget
+ * @returns {WeakMap<object,any>}
+ */
+export const makeLRUCacheMap = budget => {
+  if (!isSafeInteger(budget) || budget < 0) {
+    throw new TypeError('budget must be a safe non-negative integer number');
+  }
+  /** @type {Map<object, DoublyLinkedCell>} */
+  const map = new Map();
+  let count = 0; // count must remain <= budget
+  // As a sigil, `head` uniquely is not in the `map`.
+  const head = makeSelfCell(undefined, undefined);
+
+  const touchCell = key => {
+    const cell = map.get(key);
+    if (cell === undefined) {
+      return undefined;
+    }
+    // Becomes most recently used
+    spliceOut(cell);
+    spliceAfter(head, cell);
+    return cell;
+  };
+
+  const has = key => touchCell(key) !== undefined;
+  freeze(has);
+
+  const get = key => touchCell(key)?.value;
+  freeze(get);
+
+  const set = (key, value) => {
+    let cell = touchCell(key);
+    if (cell !== undefined) {
+      cell.value = value;
```

Comment:
This seems to be missing either an early `return` or a following `else`, without which a new cell will be created and added.

Reply:
Good catch. Fixed.